### PR TITLE
Support secrets for dataset

### LINKED ISSUE #447: [FEATURES] Support secrets for dataset
**What feature you'd like to add:**

Support Secrets for dataset. Currently, Some sensitive info like fs.oss.accessKeyId is observable by all the user. We should support secret for such scenario. 

```
apiVersion: data.fluid.io/v1alpha1
kind: Dataset
metadata:
  name: imagenet
spec:
  mounts:
  - mountPoint: oss://<OSS_BUCKET>/<OSS_DIRECTORY>/
    name: imagenet
    options:
      fs.oss.accessKeyId: <OSS_ACCESS_KEY_ID>
      fs.oss.accessKeySecret: <OSS_ACCESS_KEY_SECRET>
      fs.oss.endpoint: <OSS_ENDPOINT>
```

Take the following examples as reference:

https://kubernetes.io/zh/docs/concepts/configuration/secret/
https://help.aliyun.com/document_detail/134903.html

**Why is this feature needed:**

### PR DESCRIPTION
**Pull request by XIAO-HOU**
### Ⅰ. Describe what this PR does

Support secrets for dataset.
### Ⅱ. Does this pull request fix one issue?

fix #447 
### Ⅳ. Describe how to verify it

You can create a dataset and configure its mount options using secret. Here is an example:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: mysecret
stringData:
  fs.oss.accessKeyId: <OSS_ACCESS_KEY_ID>
  fs.oss.accessKeySecret: <OSS_ACCESS_KEY_SECRET>
---
apiVersion: data.fluid.io/v1alpha1
kind: Dataset
metadata:
  name: mypod
spec:
  mounts:
    - mountPoint: oss://<OSS_BUCKET>/<OSS_DIRECTORY>/
      name: myoss
      options:
        fs.oss.endpoint: <OSS_ENDPOINT>
      encryptOptions:
        - name: fs.oss.accessKeyId
          valueFrom:
            secretKeyRef:
              name: mysecret
              key: fs.oss.accessKeyId
        - name: fs.oss.accessKeySecret
          valueFrom:
            secretKeyRef:
              name: mysecret
              key: fs.oss.accessKeySecret
```
### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/utils/secret.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"context"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//GetSecret gets the secret.
+//It returns a pointer to the secret if successful.
```
Comment (xieydd) [inline pkg/utils/secret.go:26]: Add space after `//`

**Review by xieydd — CHANGES_REQUESTED** [review]

File: pkg/utils/secret.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+func GetSecret(client client.Client, name, namespace string) (*v1.Secret, error) {
+
```
Comment (cheyang) [inline pkg/utils/secret.go:28]: serviceAccount是否要有访问secret的权限？

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/utils/secret.go (hunk 2 above)
Comment (cheyang) [reply pkg/utils/secret.go:28]: 是不是要更新role的权限？

**Comment by cheyang** [comment]
CRD也需要重新生成吧？没有看到

File: pkg/utils/secret.go (hunk 1 above)
Comment (XIAO-HOU) [reply pkg/utils/secret.go]: 已解决，谢谢

Comment (XIAO-HOU) [reply pkg/utils/secret.go]: fixed it. thank you.

**Comment by XIAO-HOU** [comment]
> CRD也需要重新生成吧？没有看到

yiyi

**Comment by XIAO-HOU** [comment]
> CRD也需要重新生成吧？没有看到

已生成。

**Pull request merged**